agent/agentscripts/agentscripts.go

```diff
@@ -314,6 +330,36 @@ func (r *Runner) run(ctx context.Context, script codersdk.WorkspaceAgentScript)
 		} else {
 			logger.Info(ctx, fmt.Sprintf("%s script completed", logPath), slog.F("execution_time", execTime), slog.F("exit_code", exitCode))
 		}
+
+		if r.scriptCompleted == nil {
+			logger.Debug(ctx, "r.scriptCompleted unexpectedly nil")
+			return
+		}
+
+		var stage proto.Timing_Stage
+		switch option {
+		case ExecuteStartScripts:
+			stage = proto.Timing_START
+		case ExecuteStopScripts:
+			stage = proto.Timing_STOP
+		case ExecuteCronScripts:
+			stage = proto.Timing_CRON
+		}
+
+		_, err = r.scriptCompleted(ctx, &proto.WorkspaceAgentScriptCompletedRequest{
```

Comment:
Suggestion: Send asynchronously to avoid delaying access to the workspace.

Also worth considering: Send two payloads, one at start, and another on completion. This can give us useful data even in the event of agent crash or what-not.